test/Tester/LogFomatterTests.cs

```diff
@@ -14,157 +16,447 @@ public class LogFomatterTests
+                });
+            var servicesProvider = services.BuildServiceProvider();
+            servicesProvider.GetRequiredService<OptionsLogger>().LogOptions();
+
+            var logFormatters = servicesProvider.GetServices<IOptionFormatter>();
+            Assert.Equal(3, logFormatters.Count());
+            Assert.True(logFormatters.First() is TestOptionsFormatter);
+            Assert.True(logFormatters.ElementAt(1) is TestOptionsFormatter);
+            Assert.True(logFormatters.ElementAt(2) is TestOptionsFormatter);
+            var logFormatter = servicesProvider.GetService<IOptionFormatterResolver<TestOptions>>();
+            Assert.True(logFormatter is TestOptionsFormatter.Resolver);
+            var actual = servicesProvider.GetRequiredService<TestLoggerFactory>();
+            Assert.Equal(expected.ToString(), actual.ToString());
         }
 
-        public class TestOptionsFormatter2 : IOptionFormatter<TestOptions>
+        [Fact]
+        public void CustomFormatterResolverOverridesDefaultFormatter_PostRegistration()
         {
-            public string Category { get; }
+            // expected output
+            TestLoggerFactory expected = BuildNamedOptionsExpectedResult();
 
-            public string Name => nameof(TestOptions);
+            // actual output
+            var services = new ServiceCollection();
+            services.AddOptions();
+            services.AddSingleton<TestLoggerFactory>();
+            services.AddSingleton<ILoggerFactory>(sp => sp.GetRequiredService<TestLoggerFactory>());
+            services.AddSingleton(typeof(ILogger<>), typeof(TestLogger<>));
+            services.AddSingleton<OptionsLogger, TestOptionsLogger>();
+            // defaults
+            Enumerable
+                .Range(1, 2)
+                .ToList()
+                .ForEach(i =>
+                {
+                    string name = i.ToString();
+                    services.Configure<TestOptions>(name, (options => options.IntField = i));
+                    services.TryConfigureFormatter<TestOptions, TestOptionsFormatter2.Resolver>(name);
+                });
+            // post register overrides
+            services.Configure<TestOptions>("3", (options => options.IntField = 3));
+            services.ConfigureFormatter<TestOptions, TestOptionsFormatter.Resolver>("3");
+            var servicesProvider = services.BuildServiceProvider();
+            servicesProvider.GetRequiredService<OptionsLogger>().LogOptions();
+
+            var logFormatters = servicesProvider.GetServices<IOptionFormatter>();
+            Assert.Equal(3, logFormatters.Count());
+            Assert.True(logFormatters.First() is TestOptionsFormatter);
+            Assert.True(logFormatters.ElementAt(1) is TestOptionsFormatter);
+            Assert.True(logFormatters.ElementAt(2) is TestOptionsFormatter);
+            var logFormatter = servicesProvider.GetService<IOptionFormatterResolver<TestOptions>>();
```

Comment:
So named option 1 and 2 are configured and TestOptionsFormatter2, and named option 3 are configured with TestOptionsFormatter. But then they all use TestOptionsFormatter. Is this expected behavior?